src/performance/network.ts

```diff
@@ -0,0 +1,46 @@
+import { Event } from "@clarity-types/data";
+import { NetworkState } from "@clarity-types/performance";
+import config from "@src/core/config";
+import { schedule } from "@src/core/task";
+import { clearTimeout, setTimeout } from "@src/core/timeout";
+import encode from "./encode";
+
+export let state: NetworkState[] = [];
+let timeout: number = null;
+
+export function compute(entry: PerformanceResourceTiming): void {
+    // Do not instrument calls to configured upload URL
+    // This will prevent getting this code into cyclic loop:
+    //   - An upload triggers a new network request, which in turn will be captured by PerformanceObserver
+    //   - And, this new performance entry will cause code below to queue up a new event, eventually leading to another upload
+    if (config.url.length > 0 && entry.name.indexOf(config.url) >= 0) { return; }
+
+    state.push({
+        url: entry.name,
+        data: {
+            start: Math.round(entry.startTime),
+            duration: Math.round(entry.duration),
+            size: entry.transferSize,
+            target: null,
+            initiator: entry.initiatorType,
```

Comment:
Might want encoded/decoded size?

Reply:
I kept it out intentionally but we can add more fields as the need arises. Even for full URL, I am only storing the host to not go overboard. Basically transferSize is super set of encoded size. Decode is more of after compression and wasn;t sure if it was immediately helpful. 

---
In reply to: [347586302](https://github.com/microsoft/clarity-js/pull/160#discussion_r347586302) [](ancestors = 347586302)